Remove RIF from m_rifsToAdd before deleting it

tshalvi opened the pull request:
<!--
Please make sure you have read and understood the contribution guildlines:
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

1. Make sure your commit includes a signature generted with `git commit -s`
2. Make sure your commit title follows the correct format: [component]: description
3. Make sure your commit message contains enough details about the change and related tests
4. Make sure your pull request adds related reviewers, asignees, labels

Please also provide the following information in this pull request:
-->

**What I did**
I extended the RIF removal functionality to also remove the port from the m_rifsToAdd list.

**Why I did it**
Typically, the counter and object handling logic follows a strict sequence:

Create an object, then start counter polling.
Stop counter polling, then remove the object.
However, there is deferred logic for RIF counters, where counter polling starts based on a timer rather than immediately.

This process generally works as follows:
1. Create an object and add it to a list upon receiving an APP_DB update.
2. Start counter polling for all objects in the list during the timer event.
3. Stop counter polling for an object.
4. Remove the object.

If RIF creation and removal occur frequently, removal can happen before the timer event. As a result, the timer may start counter polling for an object that has just been removed, causing the following error message:
```ERR syncd#SDK: :- processFlexCounterEvent: port VID oid:0x600000000099d, was not found (probably port was removed/splitted) and will remove from counters now```

**How I verified it**
It has not been verified yet. Verification will be done during the nightly regression.

**Details if related**

tshalvi committed:
Remove RIF from m_rifsToAdd before deleting RIF

tshalvi committed:
Add newline at end of file

stephenxs commented on orchagent/intfsorch.cpp:
```diff
@@ -1319,6 +1319,15 @@ bool IntfsOrch::removeRouterIntfs(Port &port)
     const auto id = sai_serialize_object_id(port.m_rif_id);
     removeRifFromFlexCounter(id, port.m_alias);
 
+    for (auto it = m_rifsToAdd.begin(); it != m_rifsToAdd.end(); ++it)
```
a minor comment. no need to call `removeRifFromFlexCounter` if the object is in the list

tshalvi replied:
done

tshalvi committed:
Invoke removeRifFromFlexCounter only if router interface is not found in m_rifsToAdd

tshalvi committed:
missing semicolon

prsunny committed:
Merge branch 'master' into master_remove_rif_from_rif_queue_before_port_removal

saiarcot895 committed:
Fix p4orch tests after SAI update (#3337)

* Make sure p4orch tests always run, not just when gcov is enabled
* Update JSON fields in p4orch for latest SAI
* Disable the disabled-optimization for 2 ACL Manager tests in p4orch

2 test functions in acl_manager_test.cpp for p4orch are near 400 lines,
with macro expansion almost certainly adding a number of lines. On armhf
and arm64, GCC is complaining that it ran out of memory to do
optimizations here (specifically, keep track of const and copies).

liuh-80 committed:
Add mgmt VRF support. (#3299)

Add mgmt VRF support.

#### Why I did it
Fix following issue:
https://github.com/sonic-net/sonic-buildimage/issues/19638

### How I did it
Add VRF parameter and bind ZMQ to VRF.

#### How to verify it
Pass all UT.
Manually verify issue fixed.


### Description for the changelog
Add mgmt VRF support.

abdosi committed:
Fix the tlm_teamd deleting STATE_DB LAG_TABLE entry. 

What I did:
Fixes:
sonic-net/sonic-buildimage#20059

Why I did:
On T2 testbed with multiple backend port channel we have seen sometime Portchannel gets created fine and with entry in APP_DB and STATE_DB gets populated. tlm_teamd is able to get the teamdctl handle to get state dump view of teamd. However while getting dump if might have passed in 1st iteration but it might fail in 2nd iteration (transient issue in getting data using teamdctl) which result in deletion of State db entry which is not correct. Instead we should just clean up local cache and wait for retry done as part of Select Timeout cycle where we try to get dump again.

mukeshmv committed:
Add suppport for SAI DASH appliance object (#3284)

What I did
Add support for creating and deleting the new DASH appliance object in SAI.
Update tests to include the local region ID attribute in the DASH appliance table

Why I did it
New object added in SAI to populate Local region ID field.

inspurSDN committed:
fix error in rif_rates.lua (#3218)

why I did:
    the values cant't be got from COUNTERS and RATES table in counter_db as removing routing interfaces
what I did:
    checking the values are existing first, and then calculating rate values

abdosi committed:
Fix State Db LAG_MEMBER_TABLE removal not happening. (#3347)

What I did:
Original issue and and PR: #3333

Why I did:
Fix the failure in test_po_update.py as seen in this PR checker: sonic-net/sonic-buildimage#20610
Previous fix blocked State Db LAG_MEMBER_TABLE deletion which is not correct as this table is created by tlm_teamd (owner).
Intention was to prevent deletion of State Db LAG_TABLE owner of which is teamsyncd.

inspurSDN committed:
trap_rates.lua get value error (#3219)

why I did:
    If the SAI_COUNTER_STAT_PACKETS field in the COUNTERS table in the counter db is 0,
    it will be calculated as a Boolean value by trap_rates.lua, resulting in a script operation error.

what I did:
    Force in_pkts to be converted into a numerical value before operation

Pan-XT committed:
[MultiDB]:sonic-swss replace old API with new APIs (#3292)

Replace DBConnector old APIs with new APIs to enable Multi DB

I implemented similar changes in #1138

vivekrnv committed:
[DASH] Add support for ENI counters (#3266)

* Add support for DASH ENI counters

Signed-off-by: Vivek Reddy <vkarri@nvidia.com>

* Reserve the size before populating the vector

Signed-off-by: Vivek Reddy <vkarri@nvidia.com>

* Remove Flex Counter before removing the ENI object

Signed-off-by: Vivek Reddy <vkarri@nvidia.com>

* Use the dvs object from dash_db

* Add null check after metadata object type check

---------

Signed-off-by: Vivek Reddy <vkarri@nvidia.com>
Co-authored-by: Sudharsan Dhamal Gopalarathnam <sudharsand@nvidia.com>

siqbal1986 committed:
test_vnet.py refactor (#3352)

This PR refactors the test_vnet.py file. It separates the common functionality into a library file and keeps the tests in place.
What I did
Split up the test_vnet.py and put the common functionality into a library file.
Why I did it
The test_vnet.py has 29 tests. When these tests are run, it takes a log time for these tests to execute.
In addition, these tests run in the same VS container and some tests do not perform a full cleanup after the run. As a result, the next test have to deal with stale state.v The plan is to create a library so future tests can be added in a separate file

dgsudharsan committed:
Merge branch 'master' into master_remove_rif_from_rif_queue_before_port_removal

saiarcot895 committed:
Use -fsanitize=address instead of -lasan for ASAN linker flags (#3344)

What I did
For the ASAN linker flags, use -fsanitize=address instead of manually specifying -lasan.

Why I did it
This allows the compiler to use some other linkages instead of -lasan in case that's not the correct one for the platform. It (should) also make sure that the ASAN library is the first library that gets linked and loaded at runtime.
This should also fix a flaky issue with the p4orch tests failing to run because ASAN isn't loaded first (not sure why this is flaky though, it should be pretty consistent).

prgeor committed:
Handler Port oper down error status notification (#3350)

Handler Port oper down error status notification as per sai_port_error_status_t flag as part of existing port oper status notification sai_port_oper_status_notification_t

What I did
Whenever Orchagent gets the notification for port_state_change for link down event, OA will update the error count for the error flags that are set and the timestamp of the error port down event in new STATE_DB table PORT_OPERR_TABLE

Why I did it
To correlate various error events like MAC local and remote fault with the link oper down status notification.

deepak-singhal0408 committed:
VOQ: Set the ECMP group size to 128. (#3351)

What I did
Set the ECMP group size on VOQ platform to 128.

Why I did it
in SAI 11.x, the MAX ECMP group size on DNX chipsets is set to 64. This is causing issues on VOQ platforms, where ECMP group size in control plane (ebgp+ibgp) is set to 128. We see ECMP group creation failures and eventually leading to orchagent crash.
Related issue: sonic-net/sonic-buildimage#20622

vivekrnv committed:
[ACL] Add support to match on Tunnel Termination (#3320)

* [ACL] Add support to match on Tunnel Termination
Add support to match an ACL rule on Tunnel Termination Flag.
Added UT and verified create_acl_entry with this attribute is working properly

liushilongbuaa committed:
[ci] Fix reportgenerator installation error

wangxin committed:
[ci] Fix reportgenerator installation error #3369

What I did
Fix pipeline error.
here is the error message:

/tmp/823d4a0e-75c7-4ca8-b487-06e1f38eae6d/restore.csproj : error NU1202: Package dotnet-reportgenerator-globaltool 5.4.1 is not compatible with net7.0 (.NETCoreApp,Version=v7.0) / any. Package dotnet-reportgenerator-globaltool 5.4.1 supports:
/tmp/823d4a0e-75c7-4ca8-b487-06e1f38eae6d/restore.csproj : error NU1202:   - net8.0 (.NETCoreApp,Version=v8.0) / any
/tmp/823d4a0e-75c7-4ca8-b487-06e1f38eae6d/restore.csproj : error NU1202:   - net9.0 (.NETCoreApp,Version=v9.0) / any

+ reportgenerator -reporttypes:Cobertura '-reports:tests/*coverage.xml' -targetdir:.
/agent/_work/_temp/057fa35b-9e5b-426a-bd88-2c5956cb8dcf.sh: line 2: reportgenerator: command not found
Why I did it

How I verified it
pipeline in this PR didn't shows that error message.

cscarpitta committed:
[fpmsyncd]: Add support for SRv6 (#3123)

* [fpmsyncd]: Add support for SRv6
What I did
Extended fpmsyncd to process Netlink TLVs containing SRv6 information (SRv6 local SIDs and routes).
Extend fpmsyncd to process SRv6 routes and local SIDs received from FRR
Add test cases to verify SRv6 functionality

Why I did it
fpmsyncd did not process Netlink TLVs containing SRv6 information (SRv6 local SIDs and routes).

mramezani95 committed:
[mirror] erspan ipv6 underlay (see PR #1817) (#3317)

* Applied changes from PR #1817
Why I did it
Unlock erspan capability on td3 and beyond that supports erspan ipv6 encap.

What I did
Extend test_MirrorAddSetRemove, test_MirrorToVlanAddRemove, test_MirrorToLagAddRemove, test_MirrorDestMoveVlan, test_MirrorDestMoveLag to vs test erspan ipv6 encap
On td3

krismarvell committed:
platform/innovium renaming to platform/marvell-teralynx (#3252)

What I did
asic type checks for innovium platforms renamed

innovium to marvell-teralynx
Why I did it
asictype is being renamed in sonic-buildimage repo and hence respective platform checks needs changes in swss

siqbal1986 committed:
Added support for "UNDERLAY_SET_DSCP" and "UNDERLAY_SET_DSCPV6" tables (#3307)

What I did
This PR adds the logical table UNDERLAY_SET_DSCP and UNDERLAY_SET_DSCPV6.
This feature allows SONIC to match an ingress L4 packet and change the DSCP field of the outer header when the packet is egressing. These tables are only created on Cisco-8000, Mlnx and VS platforms.
The PR also adds support to check if the platform supports the following SAI attributes.
SAI_SWITCH_ATTR_ACL_USER_META_DATA_RANGE
SAI_ACL_ENTRY_ATTR_ACTION_SET_ACL_META_DATA
SAI_ACL_ENTRY_ATTR_FIELD_ACL_USER_META

theasianpianist committed:
Ensure second route group OID is correct (#3380)

What I did
When grabbing the second route group OID, ensure that it is distinct from the first route group OID

Why I did it
The order in which DB keys are returned is not deterministic, so blindly taking the last key returned will sometimes lead to selecting the wrong OID

abdosi committed:
Vlanmgrd handling of portchannel does not exist more gracefully. (#3367)

What I did:
Fixes sonic-net/sonic-buildimage#20684

Why I did:
It is seen when doing config reload on multi-asic (multiple instances of swss/teamd) their might be race condition when teammrd might be getting SIGTERM and cleaning up the port-channel in kernel and vlanmgrd might be adding the port channel as member in kernel and this can cause the exception as this operation as being done in async.

How I did:
To handle the failure gracefully if we get an exception when portchannel as vlan member in kernel we check if Portchannel exists and if not we just retry to minimize the risk of failure due to above race-condition.

stephenxs committed:
Create counter for the queue to which the host CPU traffic is sent when create_only_config_db_buffers  is enabled (#3334)

Create counter for the queue to which the host CPU traffic is sent when create_only_config_db_buffers is enabled
*There is a configuration to optimize the buffer performance DEVICE_METADATA|localhost.create_only_config_db_buffers. Using this configuration the buffer counters are created only for the queues and PGs with buffer configured for performance optimization.
However, to have counters for CPU tx queue, we have to configure it in BUFFER_QUEUE, mainly for data traffic, which is a bad coupling.
Using this PR the counter will be created for CPU tx queue without data buffer configuration.

vincentpcng committed:
Add port FEC BER feature swss, HLD#1829 (#3363)

What I did
This is to add the swss changes for the feature port FEC BER.
It modify the port_rates.lua script to compute the BER and stored them in the DB .
Two additonal PR(s) will address the cli ( sonic-utilities) and the sonic-mgmt changes

Why I did it
The HLD for this feature is HLD#1829

How I verified it
We verify the counters internally
(1) verify when link has correctable FEC errors, the BER were calcuated accordingly
(2) verify 400G with 4x100 and the BER , four serdes with 100G rate
(3) run a redis-script to polling the DB counters and verify the calculation
(4) stop polling manually modify the uncorrectable counter and verify the cli display and calculation

theasianpianist committed:
Skip route group immutable test (#3393)

What I did
Skip VS test dash/test_dash_route_group.py::test_bound_route_group_immutable

Why I did it
This test causes orchagent to crash since the VS SAI has not been updated to include implicit deletion of high-volume DASH objects (https://github.com/sonic-net/SONiC/blob/master/doc/dash/dash-sonic-hld.md#18-implicit-deletion-of-sai-objects)

vivekrnv committed:
[Dash] Handle FC update for existing ENI objects (#3373)

Why I did
When FC is enabled after being disabled, FlexCounters are not installed for existing ENI's.

What I did
Fix the flow.

a114j0y committed:
[fpmsyncd] support pipeline to flush with a timer (#3241)

What I did
change the pipeline size for RouteSync
modify the fpmsyncd flush behavior so that it could delay the flush with an expiring timer
the timer is simulated by setting the timeout for select function

mramezani95 committed:
Fixed narrowing conversion issues. (#3395)

What I did
Fixed 3 narrowing conversion issues in the code.

Why I did it
g++ 9.4.0 and 10.5.0 both reported compile errors when trying to compile aclorch.cpp using -Werror and -Wconversion flags.

mlok-nokia committed:
[Chassis][LAG_ID] Address the same lagid been used in two different LCs issue (#3303)

What I did
Create SYSTEM_LAG_IDS_FREE_LIST for assign lagId for all portchannel creation.

Portchannel creation
a) If Portchannel is created with a valid plagid
* check if plagid is in free list, use plagid and remove it from SYSTEM_LAG_IDS_FREE_LIST. Add this lagid to the SYSTEM_LAG_ID_SET for debug info
* If plagid is not in the FREE_LIST, lpop and use the first lagid from the SYSTEM_LAG_IDS_FREE_LIST. Add this lagid to the SYSTEM_LAG_ID_SET for debug info
b) If Portchannel is created with invalid plagid or without any lagid
lpop and use the first lagid from the SYSTEM_LAG_IDS_FREE_LIST. Add this lagid to the SYSTEM_LAG_ID_SET for debug info
Portchannel delection
Append the lagid to the end of SYSTEM_LAG_IDS_FREE_LIST. Also remove it from SYSTEM_LAG_ID_SET.

lolyu committed:
[dualtor] Fix standby neighbor inconsistency after warm reboot (#3356)

What I did
Let's enable mux orch to cache the neighbor updates during warm-reboot restoration, and do the updates after everything else being settled (mux configs/peer switch config are applied), so the neighbors could be reprogrammed based on the mux state.

Microsoft ADO (number only): 30148066
Signed-off-by: Longxiang Lyu lolv@microsoft.com

siqbal1986 committed:
VNET and BGP route coexistence (#3345)

What I did
This PR adds the logic to give Vnet routes precedence over BGP learnt route.
This PR also refactors the test_vnet.py to break out the common code into a library.
Why I did it
Currently if a route is learnt via BGP and added into the hardware, adding a VNET route results in failure. In addition due to a bug in VnetOrch, we start advertising the failed route, This prompts the BGP to remove the learnt route in favor of the local route. Since the Vnet Orch doesn't retry adding the Vnet route, This results in no route being present in the Hardware.

theasianpianist committed:
Only set OUTBOUND_CA_TO_PA attributes when needed (#3381)

What I did
Only set SAI_OUTBOUND_CA_TO_PA_ENTRY_ATTR_OVERLAY_DMAC and SAI_OUTBOUND_CA_TO_PA_ENTRY_ATTR_USE_DST_VNET_VNI when provided in configs

Why I did it
These attributes are required for all scenarios and will cause SAI validation issues if set when not needed.

Ndancejic committed:
[muxorch] Process nexthops using bulker during mux switchover (#3390)

Using nexthop object bulker when processing neighbors. This is only enabled during mux switchover.
When processing neighbors during mux switchover:

- add nexthop object to add/remove using nexhop bulker
- flush nexthop bulker
- post-process bulker results

saksarav-nokia committed:
Fix Voq chassis orchagent crash with 34K routes (#3329)

What I did
Don't add the remote system neighbor if the same neighbor exists.
Why I did it
The IMM has two asics and has 2 port channels in each asic and 2 port members in each port channel.
The ip address is configured on each port channel and bgp is enabled. The neighbor and routes are learned on these port channel.

krismarvell committed:
platform/marvell renaming to platform/marvell-prestera (#3378)

What I did
asic type checks for marvell platforms renamed

marvell to marvell-prestera
Why I did it
asictype is being renamed in sonic-buildimage repo and hence respective platform checks needs changes in swss

mramezani95 committed:
[vlanmgrd]: Preventing VLAN interfaces from going down when all of their member ports are down (#3370)

What I did
Started up the "dummy" interface that is already connected to the bridge named "Bridge" so that when all ports connected to the bridge are DOWN, the bridge interface remains UP. This ensures that when there is one VLAN interface connected to the bridge, it remains UP even when all of the VLAN's member ports are DOWN.
Since the VLAN interface is UP immediately after creation (because the "dummy" interface is UP), at least one route to the VLAN's subnet is added to the ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY table (one route if the VLAN is created in a peerless Vnet, and two routes if it is created in a VNet with one peer). Tests in tests/test_vnet.py have been updated to reflect this fact.

Why I did it
Before this change, when all member ports of a VLAN interface went down, the VLAN interface also went operationally down. This effectively disabled the advertisement of the VLAN interface's subnet IP to neighbors and prevented the IP-in-IP decapsulation feature for the VLAN interface from working properly.

mramezani95 committed:
Updated SWSS build and install instructions (#3396)

What I did
Updated build and install instructions in README.md and tests/README.md to include missing steps and removed outdated information.

Why I did it
sonic-swss cannot be built using current instructions in README.md.

DavidZagury committed:
Improve log information when exception is caught in Orch2 class (#3401)

What I did
Add the information regarding which class the exception is originated from when Orch2 catches an exception.

Why I did it
An Exception was thrown, and the error log given was:
ERR swss#orchagent: :- doTask: Logic error: map::at 
This gives almost no information on where the exception happened.

stepanblyschak committed:
[portsorch] query SAI_HOSTIF_ATTR_QUEUE switch capability once (#3260)

What I did

Support for SAI_HOSTIF_ATTR_QUEUE is a switch level capability, no need to query it per port.

Why I did it

As part of the goal to optimize boot time for systems with many ports. This one is the first step into that direction, extracting all non per-port operations.

liat-grozovik committed:
Merge branch 'master' into master_remove_rif_from_rif_queue_before_port_removal

Junchao-Mellanox committed:
[buffer] indicate shared pool by putting a very small shared buffer pool size (#3387)

What I did
Put a very small shared buffer pool size when shared pool is detected enabling.

Why I did it
Avoid a timing issue like following:

the buffer pool sizes, xoff have initialized to 0, which means the shared headroom pool is disabled
but the buffer profiles already indicate the shared headroom pool is enabled
later on the buffer pool sizes are updated with xoff being non-zero
In case the orchagent starts handling buffer configuration between 2 and 3, it is inconsistent between buffer pools and profiles, which fails Mellanox SAI sanity check and causes orchagent abort

prsunny committed:
Merge branch 'master' into master_remove_rif_from_rif_queue_before_port_removal

saiarcot895 committed:
Upgrade the Azure Pipeline build to use Bookworm (#3413)

What I did

Update the Azure Pipeline build to build for Bookworm.
This is dependent on sonic-net/sonic-buildimage#20697.

siqbal1986 committed:
renamed hasBgpRoute to isRouteExists in routeOrch (#3412)

What I did
Renamed the function hasBgpRoute to hasBgpRoute in routeOrch and in places it called from

mukeshmv committed:
Fix dash appliance delete error if SAI api is not implemented (#3411)

RemoteRedisSaiInterface wrongly returns a stale SAI object ID
even if the underlying SAI API object create returns NOT_IMPLEMENTED.
Fix the error caused by saving and using this stale dash appliance
object ID during subsequent object delete.

liat-grozovik committed:
Merge branch 'master' into master_remove_rif_from_rif_queue_before_port_removal

saiarcot895 committed:
Fix codeql following Bookworm upgrade (#3416)

What I did
Fix the codeql checker following the Bookworm and libnl 3.7.0 upgrade.

dgsudharsan committed:
Fixing fpmsyncd to handle scaled nexthops (#3361)

What I did
Fixing fpmsyncd to handle scaled nexthops(e.g. 256 nexthops). In this scenario the max message len is increased to accommodate zebra message size. In addition initialized nlmsg_set_default_size to allow netlink to process the increased message size as well without which libnl would crash.

Why I did it
To support increased nexthops.

stephenxs committed:
Do not start counter polling for queues with zero buffer profiles if create_only_config_db_buffers is enabled (#3360)

What I did

Do not start counter-polling for queues with zero buffer profiles if create_only_config_db_buffers is enabled.

Why I did it

If create_only_config_db_buffers is enabled, counters will be polled only on queues with a buffer configured based on the hypothesis that a customer must configure a buffer profile before using the queue.
The system performance can be optimized by reducing the number of counter polled.

However, zero buffer profiles are configured on queues of inactive ports to reclaim unused buffers. Those queues are not used by customers. Hence, we do not need to start counter polling for queues with such buffer profiles configured.

liat-grozovik committed:
Merge branch 'master' into master_remove_rif_from_rif_queue_before_port_removal

divyachandralekha committed:
PVST feature support (#3425)

Pterosaur committed:
[misc]: Add a dev folder for local development (#3368)

What I did
Add a dev folder, which includes a docker-compose environment for local development.

Why I did it
A developer has to download and install all dependences and setup the development environment if he would like to develop this submodule. This PR is for relieving the developer from the tedious setup process.

anilkpan committed:
Vlanmgr and Fdborch changes for PAC (#3143)

Added Vlanmgr and Fdborch changes to support PAC.
The following PAC operations are supported to be passed down from Vlanmgr to OA for updating the ASIC DB:

Port learning mode
FDB entry
VLAN membership

abdosi committed:
vlanmgrd not to throw exception for Portchannel ip link add because of race condition with PortChannel removal. (#3426)

* Fix ip link add command exception for portchannel being added to vlan
in case where portchannel removal can cause reace-condition with being
added as member to the vlan.

abdosi committed:
Added change not to create ECMP Group in SAI and program the route if none of ECMP members are active/link-up (#3394)

What I did:
Added change not to create ECMP Group in SAI and program the route if none of the ECMP members are active/link-up.
Also do not program the Temp Route if Neigh is not active (Link Down)

Also as part of this change if Route is not programmed and if we remove that route than decrement VRF Reference count in removeRoute as removeRoutePost will not be called in this case.

Why I did:
In scale setup of T2 it's possible all links can go down simultaneously which case we can get Route messages with all nexthops being in down state. In such case we might create empty Nexthop Group in SAI for the given route which causes not needed SAI call for Nexthop Group creation and also create traffic blackhole for the route where that route can still forward traffic via default route if eligible/applicable.

Also in this case no point to add Temp Route if neighbor is link down.

prabhataravind committed:
[dash]: Set SAI_DIRECTION_LOOKUP_ENTRY_ATTR_DASH_ENI_MAC_OVERRIDE_TYPE to dst mac (#3423)

What I did
Set SAI_DIRECTION_LOOKUP_ENTRY_ATTR_DASH_ENI_MAC_OVERRIDE_TYPE to SAI_DASH_ENI_MAC_OVERRIDE_TYPE_DST_MACby default.
This can be changed to be configurable once other vnet scenarios are onboarded on smartswitch.
Why I did it
For direction lookup to work properly for PL scenarios on smartswitch, ENI mac needs to match dst MAC.

bradh352 committed:
[orchagent] fix "parsePortConfig: Unknown field(mode)" (#3405)

When using "mode" as defined in
https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-port.yang
this warning is emitted for each interface:
```
WARNING swss#orchagent: :- parsePortConfig: Unknown field(mode): skipping ...
```

It appears this field is not needed in orchagent at all, so we need to at least recognize it as valid then skip processing.

liat-grozovik committed:
Merge branch 'master' into master_remove_rif_from_rif_queue_before_port_removal

zhixzhu committed:
Ignore failure in setPortPfcAsym if SAI_STATUS_NOT_SUPPORTED (#3430)

* ignore failure in setPortPfcAsym if SAI_STATUS_NOT_SUPPORTED
* return true on setPortPfcAsym SAI_STATUS_NOT_SUPPORTED, since false will make doPortTask keep repeating

divyachandralekha committed:
Fix stporch init exception (#3449)

What I did
Fixed stporch init exception
Why I did it
SAI retrieval of STP attributes may return either SAI_STATUS_NOT_SUPPORTED or SAI_STATUS_NOT_IMPLEMENTED. An exception has been added to handle such failures, preventing Orchagent from starting on systems where STP is not supported

gurugmarvell committed:
Capability query for MACSEC ACL attribute (#3385)

What I did
-Modified the MACsec orchestration logic to conditionally include or exclude the ability to match the SCI in ACL configurations based on the ASIC's capabilities.
-Implemented a capability check in SONiC to determine whether the SAI_ACL_TABLE_ATTR_FIELD_MACSEC_SCI attribute is supported by the ASIC, thus ensuring that neither SAI_ACL_TABLE_ATTR_FIELD_MACSEC_SCI nor SAI_ACL_ENTRY_ATTR_FIELD_MACSEC_SCI is used when unsupported.

Why I did it

The current implementation attempts to use the SAI_ACL_ENTRY_ATTR_FIELD_MACSEC_SCI attribute even when it's not supported by the underlying ASIC or driver, causing failures in the vendor's SAI/SDK code.
The capability check prevents these errors, ensuring compatibility with different hardware.

SuvarnaMeenakshi committed:
Remove update of mgmt oper status in swss (#3439)

Currently operational status of mgmt interface is not present or correct for multi-asic devices.

Why I did it
Initial PR that added mgmt oper status feature in swss: #630
sonic-net/sonic-buildimage#21245 adds a script to update oper status of management interface periodically. In doing so, we no longer need to update oper status of mgmt interface in swss.

---------

Signed-off-by: Suvarna Meenakshi <sumeenak@microsoft.com>

ram25794 committed:
Fix VRF update handling for loopback interfaces in IntfsOrch (#3461)

What I did

Addressed the scenario where an interface exists in m_syncdIntfses, but the incoming VRF does not match the updated VRF for the loopback interface. The solution involves updating the loopback’s VRF ID, decrementing the reference count for the old VRF, and incrementing the reference count for the new VRF

Why I did it

Sometimes DEL is missing in the orchagent as mentioned in the issue (#3339), leading to a scenario where, when a loopback interface is added to a non-default VRF, an ip2me route is incorrectly added to the default VRF. This issue occurs because the IntfsOrch is not handling the case where the interface exists in m_syncdIntfses, but the VRF does not match the updated VRF configuration.

saiarcot895 committed:
Add a delay between killing teamd processes (#3325)

* Add a delay between killing teamd processes

When killing 10 or more teamd processes, add a delay of 0.1 seconds
after every 10 kill signals/proceses. This is because in the LAG scale
tests (in `ecmp/inner_hashing/test_inner_hashing_lag.py` in sonic-mgmt),
it may create 100 LAGs, and when destroying them all, some of those LAGs
may fail to be properly destroyed, leaving some stale port channels
around. This seems to be because the netlink socket buffers on which the
teamd processes get notifications become full with events of the other
port channels/interfaces going down.

As a workaround, add some delays in killing the teamd processes, so that
the netlink buffers don't become full, causing messages to get dropped.

This delay was randomly chosen, and it seems to work well with 100 LAGs
on a KVM. It can probably made to be a bit more aggressive if needed
(i.e. maybe 0.05 seconds every 20 processes).

bradh352 committed:
[orchagent] Fix: ERR swss#orchagent: :- setPortPvid: pvid setting for tunnel Port_EVPN_XXX is not allowed (#3402)

* Fix: ERR swss#orchagent: :- setPortPvid: pvid setting for tunnel Port_EVPN_XXX is not allowed

Tunnel ports are always tagged and can't be created as untagged.  But
the code currently tries to set a pvid on a tunnel which is disallowed
leading to an error in the logs.

This is a simple fix to get rid of the error in the logs.  It
does not actually change behavior in any way other than getting rid
of an error thus helping debugability.

chiourung committed:
[swss] Fix for STATE_DB update for removed port during breakout (#3474)

Fix for STATE_DB update for removed port during breakout. The removed interface still exits in STATE_DB PORT_TABLE
Breakout port from 4x100G to 1x100G. It still have host_tx_ready for the removed interface in STATE_DB.

tshalvi committed:
Merge branch 'master' into master_remove_rif_from_rif_queue_before_port_removal

vivekrnv committed:
[ACL] Add Tunnel Next Hop redirect support (#3399)

* [ACL] Add Tunnel Next Hop redirect support

* What I did

Add support for Tunnel Next Hop as the Redirect Action in ACL Rules
Supported schema for REDIRECT_ACTION field : <ip>@<tunnel_name> (Eg: 20.0.0.3@tunnel0)
Add a new Macro in the SAI mocking infra to clear extern mock_fns set

dgsudharsan committed:
Merge branch 'master' into master_remove_rif_from_rif_queue_before_port_removal

cscarpitta committed:
[tests]: Fix SRv6 test cases (#3486)

* [tests]: Fix a typo in SRv6 locator prefix
The current SRv6 test cases have a typo in the prefix length of the locator. This PR fixes the prefix length.

prsunny committed:
Merge branch 'master' into master_remove_rif_from_rif_queue_before_port_removal
